### PR TITLE
[Storage] Support private endpoint connection apis on new api version 2019-06-01

### PR DESCRIPTION
**Pull request by priyapansrp**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by Juliehzl** [comment]
Hi @priyapansrp, you also need to do some change to make CI pass.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/StorageAccountPutPrivateEndpointConnection.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
```
[the remark this answers is not in the document]
Comment (priyapansrp) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/examples/StorageAccountPutPrivateEndpointConnection.json]: Added it.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json [hunk 2]
```diff
@@ -2381,6 +2502,138 @@
...
+          "type": "string",
+          "description": "Get the provisioning State of the private endpoint connection at the time operation was called.",
+          "enum": [
+            "Creating",
+            "Succeeded",
+            "Failed",
+            "Deleting"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
```
[the remark this answers is not in the document]
Comment (priyapansrp) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json]: Fixed the conflicting enum.

**Review by blueww — APPROVED** [review]
LGTM

One thing need notice, we have another PR on 2019-04-01. 
This PR need re copy 2019-06-01 from 2019-04-01, when https://github.com/Azure/azure-rest-api-specs/pull/7506 is merged

**Comment by blueww** [comment]
The swagger PR on 2019-04-01 is merged: https://github.com/Azure/azure-rest-api-specs/pull/7506
Please re copy 2019-06-01 from 2019-04-01 in this PR.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json [hunk 3]
```diff
@@ -734,6 +734,164 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateEndpointConnectionName"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (KrisBash) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json]: Please add a default response that implements the common error contract. This should be on all API responses, but we can start with net new ones. Here is an example: https://github.com/Azure/azure-rest-api-specs/blob/14464cf0bec27f838f1649f562c9f98cc4074855/specification/batch/resource-manager/Microsoft.Batch/stable/2019-08-01/BatchManagement.json#L105

**Review by KrisBash — CHANGES_REQUESTED** [review]
APIs look good, but all responses should include "default" w/ common error contract.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json (hunk 3 above)
Comment (priyapansrp) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json]: > Please add a default response that implements the common error contract. This should be on all API responses, but we can start with net new ones. Here is an example:
> 
> https://github.com/Azure/azure-rest-api-specs/blob/14464cf0bec27f838f1649f562c9f98cc4074855/specification/batch/resource-manager/Microsoft.Batch/stable/2019-08-01/BatchManagement.json#L105

Added default error response for these new apis.

**Comment by priyapansrp** [comment]
> The swagger PR on 2019-04-01 is merged: #7506
> Please re copy 2019-06-01 from 2019-04-01 in this PR.

I would send a new PR to sync this apiversion with previous api-version.

**Review by KrisBash — APPROVED** [review]
LGTM - thanks

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 4]
```diff
@@ -0,0 +1,1368 @@
...
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BlobServiceProperties"
+            },
+            "description": "The properties of a storage account’s Blob service, including properties for Storage Analytics and CORS (Cross-Origin Resource Sharing) rules."
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: No default responses here as @KrisBash  said.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 5]
```diff
@@ -0,0 +1,1368 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/BlobServicesName"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: The same as above.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 6]
```diff
@@ -0,0 +1,1368 @@
...
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "type": "string",
+            "description": "Optional. When specified, only container names starting with the filter will be listed."
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: The same as above.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 7]
```diff
@@ -0,0 +1,1368 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 8]
```diff
@@ -0,0 +1,1368 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 9]
```diff
@@ -0,0 +1,1368 @@
...
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 10]
```diff
@@ -0,0 +1,1368 @@
...
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/LegalHold"
+            },
+            "description": "The LegalHold property that will be set to a blob container."
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 11]
```diff
@@ -0,0 +1,1368 @@
...
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/LegalHold"
+            },
+            "description": "The LegalHold property that will be clear from a blob container."
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 12]
```diff
@@ -0,0 +1,1368 @@
...
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "description": "The entity state (ETag) version of the immutability policy to update. A value of \"*\" can be used to apply the operation only if the immutability policy already exists. If omitted, this operation will always be applied."
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 13]
```diff
@@ -0,0 +1,1368 @@
...
+            "name": "parameters",
+            "in": "body",
+            "schema": {
+              "$ref": "#/definitions/LeaseContainerRequest"
+            },
+            "description": "Lease Container request body."
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 14]
```diff
@@ -0,0 +1,1368 @@
...
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": true,
+            "type": "string",
+            "description": "The entity state (ETag) version of the immutability policy to update. A value of \"*\" can be used to apply the operation only if the immutability policy already exists. If omitted, this operation will always be applied."
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 15]
```diff
@@ -0,0 +1,1368 @@
...
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 16]
```diff
@@ -0,0 +1,1368 @@
...
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Still no default response.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json [hunk 17]
```diff
@@ -0,0 +1,1368 @@
...
+        "publicAccess": {
+          "type": "string",
+          "enum": [
+            "Container",
+            "Blob",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "PublicAccess",
+            "modelAsString": false
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Why not support modelAsString here?

**Comment by Juliehzl** [comment]
Approved in general. But @priyapansrp I hope you can raise a PR to fix these responses or create an issue to track these changes required for your APIs.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json (hunk 4 above)
Comment (priyapansrp) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json]: Here is what @KrisBash said. "Please add a default response that implements the common error contract. This should be on all API responses, but we can start with net new ones.". I will add a seperate PR to fix all existing apis.

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**